File: app/src/main/java/in/engineerakash/covid19india/util/ObservableObject.java
package in.engineerakash.covid19india.util;

import java.util.Observable;

public class ObservableObject extends Observable {

    private static ObservableObject instance = new ObservableObject();

    private ObservableObject() {
    }

    public static ObservableObject getInstance() {
        return instance;
    }

    public void updateValue(Object data) {
        synchronized (this) {
            setChanged();
            notifyObservers(data);
        }
    }
}

File: app/src/main/java/in/engineerakash/covid19india/util/ObservableObject.kt
package `in`.engineerakash.covid19india.util

import java.util.*

class ObservableObject private constructor() : Observable() {

    fun updateValue(data: Any?) {
        synchronized(this) {
            setChanged()
            notifyObservers(data)
        }
    }

    companion object {
        @JvmStatic
        val instance = ObservableObject()
    }
}